### PR TITLE
Add SPI/TWI stability test

### PR DESCRIPTION
**Pull request by nordic-segl**
NRFX-6286: [Verification] Prepare POC for testing serial interface
Prepare test for serial interface (SPI/TWI) that can be used for stability testing in climate chamber.

### CODE REVIEW

File: tests/benchmarks/i2c_endless/testcase.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  harness: console
+  harness_config:
+    fixture: gpio_i2c_bridge_two_boards
+  build_only: true
+  platform_allow:
+    - nrf52840dk/nrf52840
+    - nrf5340dk/nrf5340/cpuapp
+    - nrf54l15pdk/nrf54l15/cpuapp
+    - nrf54h20dk/nrf54h20/cpuapp
+
```
Comment (katgiadla) [inline tests/benchmarks/i2c_endless/testcase.yaml:13]: Please add obligatory `intergation_platforms` section

File: tests/benchmarks/spi_endless/testcase.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  tags: drivers spi
+  harness: console
+  harness_config:
+    fixture: gpio_spi_bridge_two_boards
+  build_only: true
+  platform_allow:
+    - nrf52840dk/nrf52840
+    - nrf54l15pdk/nrf54l15/cpuapp
+    - nrf54h20dk/nrf54h20/cpuapp
+
```
Comment (katgiadla) [inline tests/benchmarks/spi_endless/testcase.yaml:12]: Please add obligatory `intergation_platforms` section too.

**Review by katgiadla — CHANGES_REQUESTED** [review]

File: tests/benchmarks/spi_endless/testcase.yaml (hunk 2 above)
Comment (nordic-segl) [reply tests/benchmarks/spi_endless/testcase.yaml]: Done

File: tests/benchmarks/i2c_endless/src/main.c [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+	COND_CODE_1(DT_NODE_HAS_PROP(NODE_TWIS, memory_regions),                           \
+		    (__attribute__((__section__(                                               \
+			    LINKER_DT_NODE_REGION_NAME(DT_PHANDLE(NODE_TWIS, memory_regions)))))), \
+		    ())
+
+static bool test_pass;
+static int counter;
+
+#if CONFIG_ROLE_HOST == 1
+static const struct device *spim_dev = DEVICE_DT_GET(DT_BUS(NODE_SENSOR));
```
Comment (nordic-piks) [inline tests/benchmarks/i2c_endless/src/main.c:47]: spim_dev? in i2c test?

File: tests/benchmarks/i2c_endless/src/main.c [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+		test_pass = true;
+
+		/* Generate pseudo random tx_data for current test */
+		for (int i = 0; i < DATA_FIELD_LEN; i++) {
+			i2c_twim_buffer[i] = acc;
+			acc += DELTA;
+		}
+
+		/* Send data */
+		ret = i2c_write(spim_dev, i2c_twim_buffer, DATA_FIELD_LEN,
```
Comment (nordic-piks) [inline tests/benchmarks/i2c_endless/src/main.c:165]: spim_dev?

File: tests/benchmarks/i2c_endless/src/main.c [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+#define I2C_S_INSTANCE 2
+#elif CONFIG_NRFX_TWIS22
+#define I2C_S_INSTANCE 22
+#elif CONFIG_NRFX_TWIS131
+#define I2C_S_INSTANCE 131
+#else
+#error "TWIS instance not enabled or not supported"
+#endif
+
+#define NODE_SENSOR DT_NODELABEL(sensor)
```
[the remark this answers is not in the document]
Comment (nordic-segl) [reply tests/benchmarks/i2c_endless/src/main.c:30]: I can change to NODE_SLAVE :smile: if You are not afraid of inclusive language.
SPIS on other board is called here a sensor.

File: tests/benchmarks/i2c_endless/src/main.c (hunk 3 above)
Comment (nordic-segl) [reply tests/benchmarks/i2c_endless/src/main.c:47]: copy-paste error. Fixed

File: tests/benchmarks/i2c_endless/src/main.c (hunk 4 above)
Comment (nordic-segl) [reply tests/benchmarks/i2c_endless/src/main.c:165]: copy-paste error. Fixed

**Review by katgiadla — APPROVED** [review]

**Pull request merged**